internal/expr/prepare.go

```diff
@@ -175,6 +181,16 @@ func (pe *ParsedExpr) Prepare(args ...any) (expr *PreparedExpr, err error) {
 
 	// Generate and save reflection info.
 	for _, arg := range args {
+		if arg == nil {
+			return nil, fmt.Errorf("need struct, got nil")
+		}
+		switch k := reflect.TypeOf(arg).Kind(); k {
+		case reflect.Struct:
+		case reflect.Pointer:
+			return nil, fmt.Errorf("need struct, got pointer to %s. Prepare takes structs by value as they are only used for their type information", reflect.TypeOf(arg).Elem().Kind())
```

Comment:
This case is slightly different since we only deal with the type of `arg` rather than the value. Even though getting the type a second time with `reflect.TypeOf(arg).Elem().Kind()` is pretty inefficient I was thinking its ok since it only happens when the error is thrown.